webapp/src/index.js

```diff
@@ -23,13 +23,17 @@ const activityTimeout = 60 * 60 * 1000; // 1 hour
 
 class PluginClass {
     async initialize(registry, store) {
+        const userLoggedIn = isLoggedIn(store.getState());
+        let settings = false;
         registry.registerReducer(Reducer);
         Client.setServerRoute(getServerRoute(store.getState()));
 
-        const {data: settings} = await getSettings(store.getState);
-        await getConnected(true)(store.dispatch, store.getState);
+        if (userLoggedIn) {
+            settings = await getSettings(store.getState);
+            await getConnected(true)(store.dispatch, store.getState);
+        }
 
-        if (settings && settings.left_sidebar_enabled) {
+        if (!userLoggedIn || (settings !== false && settings.left_sidebar_enabled)) {
             registry.registerLeftSidebarHeaderComponent(SidebarHeader);
             registry.registerBottomTeamSidebarComponent(TeamSidebar);
         }
```

Comment:
This change in the PR is more of a bandaid than a proper fix. A few observations of the existing code:

- This `initialize` function is the only time `getSettings` is called, and its result is not stored in the redux store, so this result is sort of silo'd in this function. If any other piece of code wants to know about `settings.left_sidebar_enabled`, they will have to use `getSettings` to contact the server again. Also note that only one field is provided in the response of this endpoint, which is `settings.left_sidebar_enabled`. So this call is not very useful.

- However `getConnected` is called several times throughout the lifecycle of the plugin. The `getConnected` endpoint returns a lot of useful data. It returns user settings, but not plugin settings.

https://github.com/mattermost/mattermost-plugin-github/blob/master/server/plugin/api.go#L489-L496

I propose we change the struct to include `PluginSettings`, and rename `Settings` to `UserSettings`. We will also want to make similar name changes in [webapp/src/reducers/index.js](https://github.com/mattermost/mattermost-plugin-github/blob/master/webapp/src/reducers/index.js), where we'll change `settings` to `userSettings`, and add a new reduceer called `pluginSettings`.


Now that we have the plugin settings in the redux store, it can be used by our components. We can edit [webapp/src/components/sidebar_header/index.js](https://github.com/mattermost/mattermost-plugin-github/blob/a60ac1934e08e89e49535b9822f47352584569db/webapp/src/components/sidebar_header/index.js#L8) and [webapp/src/components/team_sidebar/index.js](https://github.com/mattermost/mattermost-plugin-github/blob/a60ac1934e08e89e49535b9822f47352584569db/webapp/src/components/team_sidebar/index.js#L8) like so to use this value. Bonus points if this can be fetched from a selector instead of accessing state this way:

```js
function mapStateToProps(state) {
    const members = state.entities.teams.myMembers || {};
    
    const sidebarEnabled = state[`plugins-${pluginId}`].pluginSettings.left_sidebar_enabled;
    const show = sidebarEnabled && Object.keys(members).length <= 1;
  
    return {
        show,
    };
}
```


Then we can get rid of the if conditional `if (!userLoggedIn || (settings !== false && settings.left_sidebar_enabled)` and just always register those two components. This gives the responsibility to the components for determining whether to _render_ the components instead of register.

Also, I think we should call `getConnected` regardless of the value of `state.entities.users.currentUserId`. The webapp could change its logic around that at anytime. It's okay if we call `getConnected` and the user isn't logged in, even if it doesn't have any useful effect. It keeps the code simpler.

Reply:
@mickmister Thank you so much for such a neat approach, will go with that.😊

Reply:
@mickmister done.